core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -482,6 +499,34 @@ object UnitActions {
         )
     }
 
+    private fun addRepairAction(unit: MapUnit, actionList: ArrayList<UnitAction>) {
+        if (!unit.hasUniqueToBuildImprovements) return
+        if (unit.isEmbarked()) return
+        val tile = unit.getTile()
+        if (!(tile.improvementIsPillaged || tile.roadIsPillaged)) return
```

Comment:
Don't you have a function `tile.isPillaged()` to check for exactly this?

Reply:
Haha, yup. What I get for merging the Road code on top